nodejs/eks/cluster.ts

```diff
@@ -147,6 +147,65 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         }, { parent: parent });
     }
 
+    // Create an AWS IAM Entity that will be the IAM entity which is automatically granted system:masters permissions in the cluster's RBAC configuration
+    let eksCreatorEntity: aws.iam.Role | undefined;
+    let eksCreatorEntityARN: pulumi.Input<aws.ARN> | undefined;
+    let eksCreatorEntityPolicy: aws.iam.Policy;
+    let eksCreatorEntityRolePolicyAttachment: aws.iam.RolePolicyAttachment;
+    let eksCreatorEntityProvider: aws.Provider | undefined;
+
+    if (args.useRoleForClusterCreation) {
```

Comment:
This looks good - can we capture this block in a new func like `createIAMProvider` or some such to simplify legibility of the class?